gluten-substrait/src/main/scala/org/apache/gluten/execution/BasicPhysicalOperatorTransformer.scala

```diff
@@ -210,14 +210,6 @@ abstract class ProjectExecTransformerBase(val list: Seq[NamedExpression], val in
   override def doTransform(context: SubstraitContext): TransformContext = {
     val childCtx = child.asInstanceOf[TransformSupport].transform(context)
     val operatorId = context.nextOperatorId(this.nodeName)
-    if ((list == null || list.isEmpty) && childCtx != null) {
-      // The computing for this project is not needed.
-      // the child may be an input adapter and childCtx is null. In this case we want to
-      // make a read node with non-empty base_schema.
-      context.registerEmptyRelToOperator(operatorId)
-      return childCtx
-    }
```

Comment:
`context.registerEmptyRelToOperator(operatorId)` looks to be suitable only when the operator simply outputs all the inputs it receives. Which is not the case here for `list.isEmpty`